packages/react-vapor/src/components/limit/Limit.tsx

```diff
@@ -0,0 +1,142 @@
+import classNames from 'classnames';
+import * as React from 'react';
+import {Input} from 'react-vapor';
+
+export interface ILimitOwnProps {
+    title: string;
+    limitLabel?: string;
+    usageNumber?: number;
+    limitNumber?: number;
+    isLimitModifiable?: boolean;
+    isLimitTheGoalToReach?: boolean;
+    isHistoryIncluded?: boolean;
+}
+
+export interface ILimitDispatchProps {
+    onRender?: () => void;
+    onDestroy?: () => void;
+}
+
+export interface ILimitProps extends ILimitOwnProps, ILimitDispatchProps {}
+
+export class Limit extends React.Component<ILimitProps, {}> {
```

Comment:
you can use a PureComponent instead of a Component. Its not a big deal but the PureComponent its a bit more faster
`React.Component<ILimitProps>`

If you want too, you can convert the component with a Functional component. Its faster than the PureComponent and give the possibility to use the useEffect.
nice doc: https://www.robinwieruch.de/react-function-component

Reply:
Functional component :heart: